server/server.go

```diff
@@ -198,11 +199,15 @@ func (s *Server) HTTPHandler() http.Handler {
 	clock := clockwork.NewRealClock()
 	mux := http.NewServeMux()
 	mux.HandleFunc(httpPathDiscovery, handleDiscoveryFunc(s.ProviderConfig()))
-	mux.HandleFunc(httpPathAuth, handleAuthFunc(s, s.Connectors, s.LoginTemplate))
+	mux.HandleFunc(httpPathAuth, handleAuthFunc(s, s.Connectors, s.LoginTemplate, s.EnableRegistration))
 	mux.HandleFunc(httpPathToken, handleTokenFunc(s))
 	mux.HandleFunc(httpPathKeys, handleKeysFunc(s.KeyManager, clock))
 	mux.Handle(httpPathHealth, makeHealthHandler(checks))
-	mux.HandleFunc(httpPathRegister, handleRegisterFunc(s))
+
+	if s.EnableRegistration {
```

Comment:
SWEET! This is a nice way to ensure that registration can't happen - it isn't even ever wired up!